server/src/termdb.server.init.js

```diff
@@ -454,10 +454,14 @@ export function server_init_db_queries(ds) {
 		return supportedChartTypes
 	}
 
-	q.getSingleSampleData = function (sampleId) {
-		const sql = cn.prepare(
-			`select term_id, value from anno_categorical where sample=? union all select term_id, value from anno_float where sample=? union all  select term_id, value from anno_integer where sample=?`
-		)
+	q.getSingleSampleData = function (sampleId, term_ids) {
+		const ids = `('${term_ids.join("','")}')`
+
+		const query = `select term_id, value from anno_categorical where sample=? and term_id in ${ids}
```

Comment:
@airenzp Should not put variables directly in the sql string, should use the interpolated values as listed in the array argument to `sql.all(...)`. This is vulnerable to sql-injection attack.

Reply:
Should be solved now.